src/main/java/com/artipie/asto/LoggingStorage.java

```diff
@@ -138,13 +138,13 @@ public LoggingStorage(final Level level, final Storage storage) {
     }
 
     @Override
-    public CompletableFuture<Transaction> transaction(final List<Key> keys) {
-        return this.storage.transaction(keys).thenApply(
+    public CompletionStage<Void> exclusively(
```

Comment:
@olegmoz maybe we can use `<T>` instead of `<Void>` here with `Function<Storage, CompletionStage<T>>` to return some result from storage if needed?